### ISSUE TITLE
FirestoreError Missing or insufficient permissions

### ISSUE THREAD

**Issue by ralphsmith80** [prescottprue/cypress-firebase#92]
I'm running locally with a user that can authenticate. I can login with that user and initiate firestore listeners without issue. However, when I do so via cypress I'm getting the following error.

> FirebaseError: Missing or insufficient permissions. at new FirestoreError

I know I have the Firestore rules are set properly because I have multiple users that work and my test user also works if I'm not using cypress. 

Here an example of the listener code. It's likely not very helpful as it's basically what you would see in the docs. 
```
 db.doc(`users/${firebaseUser.uid}`).onSnapshot(
  doc => {
    const data = doc.data()
    if (!doc.exists) {
      user = null
    }
    console.log('worked')
  },
  err => {
    // This is where it's catching
    console.error(err)
  }
)
```

I have two event listeners similar to this. Something that is a bit odd about this is that one listener will succeed and then fail while the second will always fail. 

Have you used this with Firestore listeners before?
Have you had to do anything with the Firestore rules to enable this specifically with Cypress?
Any other ideas?

**Comment by prescottprue** [comment]
Can you provide a link to your repo so Cypress setup can be confirmed? How are you calling `cy.login` in the tests? Are you passing a UID or generating the JWT before running tests?

**Comment by ralphsmith80** [comment]
I'm generating the JWT before running tests. So in that case I'm calling `cy.login` via `cy.login()`. I did try using `cy.login(Cypress.env('TEST_UID'))` without luck.

Login is not the problem here though. That works as expected. The problem is that I'm getting false errors from the [realtime listeners](https://firebase.google.com/docs/firestore/query-data/listen). 

Logging in with that user on a production server or locally works as expected but it fails when running via Cypress. 

...

I think I figured it out. 

In `cypress.json` I was disabling Chrome web security: `"chromeWebSecurity": false`

I don't recall if I need that in that past when testing against a production server but it all appears to be working now. I'm guessing firebase was blocking the request to an unsecure browser instance.
